core/include/cubos/core/net/udp_socket.hpp

```diff
@@ -0,0 +1,79 @@
+/// @file
+/// @brief Class @ref cubos::core::net::UdpSocket.
+/// @ingroup core-net
+
+#pragma once
+
+#include <cstdint>
+#include <optional>
+#include <string>
+
+#include <cubos/core/api.hpp>
+#include <cubos/core/net/address.hpp>
+
+namespace cubos::core::net
+{
+    /// @brief Represents a UDP socket.
+    /// @ingroup core-net
+    class CUBOS_CORE_API UdpSocket
+    {
+    public:
+        /// @brief Represents a result code.
+        enum class ResultCode
+        {
+            Success,
+            BindError,
+            SendError,
+            ReceiveError,
+            UnknownError
+        };
```

Comment:
Since you basically only have one result code for each operation, you might as well remove this enum and just return a boolean in each method, indicating whether the operation was successful or not.

Reply:
Ok. Just wanted a review so I can start implementing the other network utilities (`TcpSocket`, etc..) which should be similar to this one.